pkg/embed/generator.go

```diff
@@ -0,0 +1,131 @@
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/download"
+
+	"github.com/code-ready/crc/pkg/crc/machine/hyperkit"
+	"github.com/code-ready/crc/pkg/crc/machine/libvirt"
+)
+
+const (
+	blobFileName string = "blob"
+)
+
+// Define vars for build template
+var conv = map[string]interface{}{"conv": fmtByteSlice}
+var tmpl = template.Must(template.New("").Funcs(conv).Parse(`package embed
+
+// Code generated by go generate; DO NOT EDIT.
+
+func init() {
+    {{- range $name, $file := . }}
+        Add("{{ $name }}", []byte{ {{ conv $file }} })
+    {{- end }}
+}`),
+)
+
+func fmtByteSlice(s []byte) string {
+	builder := strings.Builder{}
+
+	for _, v := range s {
+		builder.WriteString(fmt.Sprintf("%d,", int(v)))
+	}
+
```

Comment:
This could be
```go
+       for i, v := range s {
+               if i%32 == 0 {
+                       builder.WriteString("\n")
+               }
+               builder.WriteString(fmt.Sprintf("%#02x,", int(v)))
        }
```
which gives a generated file which is much nicer to read.

Reply:
Sure I just hope it doesn't break anything, will try.

Reply:
Actually, using `[]byte("\x00\x00\x00...")` makes the build resource consumption much much better than `[]byte{ 0x00, 0x00, 0x00 ...}` so we should use this instead:
```
diff --git a/pkg/embed/generator.go b/pkg/embed/generator.go
index 8fdb1a1f..c3668445 100644
--- a/pkg/embed/generator.go
+++ b/pkg/embed/generator.go
@@ -32,7 +32,7 @@ var tmpl = template.Must(template.New("").Funcs(conv).Parse(`package embed
 
 func init() {
     {{- range $name, $file := . }}
-        Add("{{ $name }}", []byte{ {{ conv $file }} })
+        Add("{{ $name }}", []byte("{{ conv $file }}"))
     {{- end }}
 }`),
 )
@@ -41,7 +41,7 @@ func fmtByteSlice(s []byte) string {
 	builder := strings.Builder{}
 
 	for _, v := range s {
-		builder.WriteString(fmt.Sprintf("%d,", int(v)))
+		builder.WriteString(fmt.Sprintf("\\x%02x", int(v)))
 	}
 
 	return builder.String()
```

Reply:
The corresponding golang bug i s https://github.com/golang/go/issues/6643